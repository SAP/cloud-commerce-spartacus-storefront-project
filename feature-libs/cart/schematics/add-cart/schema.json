{
  "$schema": "http://json-schema.org/schema",
  "id": "CartSchematics",
  "title": "Cart Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the Saved cart feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": ["Saved-cart"],
        "type": "string"
      },
      "default": ["Saved-cart"],
      "x-prompt": "Which features would you like to setup?"
    }
  },
  "required": []
}
