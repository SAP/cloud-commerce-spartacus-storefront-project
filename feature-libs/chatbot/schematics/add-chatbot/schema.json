{
  "$schema": "http://json-schema.org/schema",
  "$id": "ChatBotSchematics",
  "title": "ChatBot Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the chat bot features.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["ChatBot"]
    }
  },
  "required": []
}
