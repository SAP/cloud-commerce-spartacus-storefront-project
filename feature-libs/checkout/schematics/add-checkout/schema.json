{
  "$schema": "http://json-schema.org/schema",
  "$id": "CheckoutSchematics",
  "title": "Checkout Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the Checkout feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": [
          "Checkout",
          "Checkout-B2B",
          "Checkout-Scheduled-Replenishment"
        ],
        "type": "string"
      },
      "default": ["Checkout"],
      "x-prompt": {
        "message": "Which features would you like to set up from the Checkout library?",
        "type": "list",
        "items": [
          {
            "value": "Checkout",
            "label": "Checkout"
          },
          {
            "value": "Checkout-B2B",
            "label": "Checkout (b2b features)"
          },
          {
            "value": "Checkout-Scheduled-Replenishment",
            "label": "Checkout Scheduled Replenishment (b2b feature)"
          }
        ]
      }
    }
  },
  "required": []
}
