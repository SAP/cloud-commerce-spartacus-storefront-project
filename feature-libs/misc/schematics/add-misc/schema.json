{
  "$schema": "http://json-schema.org/schema",
  "id": "MiscSchematics",
  "title": "Misc Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the misc features.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": ["Storefinder"],
        "type": "string"
      },
      "default": ["Storefinder"],
      "x-prompt": "Which features would you like to setup?"
    }
  },
  "required": []
}
