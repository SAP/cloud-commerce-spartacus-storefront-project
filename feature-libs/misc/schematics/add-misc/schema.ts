export interface Schema {
  project: string;
  lazy: boolean;
  features: string[];
}
