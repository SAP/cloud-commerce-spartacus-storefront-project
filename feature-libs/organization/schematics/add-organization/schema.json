{
  "$schema": "http://json-schema.org/schema",
  "id": "OrganizationSchematics",
  "title": "Organization Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    }
  },
  "required": []
}
