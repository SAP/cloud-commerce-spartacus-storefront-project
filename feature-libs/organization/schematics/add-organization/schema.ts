export interface Schema {
  project: string;
  lazy: boolean;
}
