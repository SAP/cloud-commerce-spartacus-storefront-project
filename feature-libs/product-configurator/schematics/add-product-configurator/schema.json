{
  "$schema": "http://json-schema.org/schema",
  "id": "ProductConfiguratorSchematics",
  "title": "Product Configurator Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the product configurator features.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": ["CPQ configurator", "Textfield configurator"],
        "type": "string"
      },
      "x-prompt": "Which product configurator features besides variant configurator would you like to setup?"
    }
  },
  "required": []
}
