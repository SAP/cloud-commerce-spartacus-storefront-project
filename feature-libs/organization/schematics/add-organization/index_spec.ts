/// <reference types="jest" />

import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import {
  LibraryOptions as SpartacusOrganizationOptions,
  SpartacusOptions,
  SPARTACUS_CONFIGURATION_MODULE,
  SPARTACUS_SETUP,
} from '@spartacus/schematics';
import * as path from 'path';
import {
  CLI_ADMINISTRATION_FEATURE,
  CLI_ORDER_APPROVAL_FEATURE,
} from '../constants';

const collectionPath = path.join(__dirname, '../collection.json');
const administrationFeatureModulePath =
  'src/app/spartacus/features/organization/organization-administration-feature.module.ts';
const orderApprovalFeatureModulePath =
  'src/app/spartacus/features/organization/organization-order-approval-feature.module.ts';
const scssFilePath = 'src/styles/spartacus/organization.scss';

// TODO: Improve tests after lib-util test update
describe('Spartacus Organization schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner('schematics', collectionPath);

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
  };

  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    configuration: 'b2c',
    lazy: true,
    features: [],
  };

  const defaultFeatureOptions: SpartacusOrganizationOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [CLI_ADMINISTRATION_FEATURE, CLI_ORDER_APPROVAL_FEATURE],
  };

  beforeEach(async () => {
    schematicRunner.registerCollection(
      '@spartacus/schematics',
      '../../projects/schematics/src/collection.json'
    );

    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'workspace',
        workspaceOptions
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'application',
        appOptions,
        appTree
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@spartacus/schematics',
        'ng-add',
        { ...spartacusDefaultOptions, name: 'schematics-test' },
        appTree
      )
      .toPromise();
  });

  describe('when no features are selected', () => {
    beforeEach(async () => {
      appTree = await schematicRunner
        .runSchematicAsync(
          'ng-add',
          { ...defaultFeatureOptions, features: [] },
          appTree
        )
        .toPromise();
    });

    it('should not install administration nor order-approval features', () => {
      expect(appTree.exists(administrationFeatureModulePath)).toBeFalsy();
      expect(appTree.exists(orderApprovalFeatureModulePath)).toBeFalsy();
    });
  });

  describe('Administration feature', () => {
    describe('styling', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', defaultFeatureOptions, appTree)
          .toPromise();
      });

      it('should create a proper scss file', () => {
        const scssContent = appTree.readContent(scssFilePath);
        expect(scssContent).toMatchSnapshot();
      });

      it('should update angular.json', async () => {
        const content = appTree.readContent('/angular.json');
        expect(content).toMatchSnapshot();
      });
    });

    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync(
            'ng-add',
            { ...defaultFeatureOptions, lazy: false },
            appTree
          )
          .toPromise();
      });

      it('should import appropriate modules', async () => {
        const administrationModule = appTree.readContent(
          administrationFeatureModulePath
        );
        expect(administrationModule).toContain(
          `import { AdministrationRootModule } from "@spartacus/organization/administration/root";`
        );
        expect(administrationModule).toContain(
          `import { AdministrationModule } from "@spartacus/organization/administration";`
        );
      });

      it('should not contain lazy loading syntax', async () => {
        const administrationModule = appTree.readContent(
          administrationFeatureModulePath
        );
        expect(administrationModule).not.toContain(
          `import('@spartacus/organization/administration').then(`
        );
      });
    });

    describe('lazy loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', defaultFeatureOptions, appTree)
          .toPromise();
      });

      it('should import AdministrationRootModule and contain the lazy loading syntax', async () => {
        const administrationModule = appTree.readContent(
          administrationFeatureModulePath
        );
        expect(administrationModule).toContain(
          `import { AdministrationRootModule, ORGANIZATION_ADMINISTRATION_FEATURE } from "@spartacus/organization/administration/root";`
        );
        expect(administrationModule).toContain(
          `import('@spartacus/organization/administration').then(`
        );
      });

      it('should not contain the AdministrationModule import', () => {
        const administrationModule = appTree.readContent(
          administrationFeatureModulePath
        );
        expect(administrationModule).not.toContain(
          `import { AdministrationModule } from "@spartacus/organization/administration";`
        );
      });
    });

    describe('i18n', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', defaultFeatureOptions, appTree)
          .toPromise();
      });

      it('should import the i18n resource and chunk from assets', async () => {
        const administrationModule = appTree.readContent(
          administrationFeatureModulePath
        );
        expect(administrationModule).toContain(
          `import { organizationTranslationChunksConfig, organizationTranslations } from "@spartacus/organization/administration/assets";`
        );
      });
      it('should provideConfig', async () => {
        const administrationModule = appTree.readContent(
          administrationFeatureModulePath
        );
        expect(administrationModule).toContain(
          `resources: organizationTranslations,`
        );
        expect(administrationModule).toContain(
          `chunks: organizationTranslationChunksConfig,`
        );
      });
    });
  });

  describe('Order approval feature', () => {
    describe('styling', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', defaultFeatureOptions, appTree)
          .toPromise();
      });

      it('should create a proper scss file', () => {
        const scssContent = appTree.readContent(scssFilePath);
        expect(scssContent).toMatchSnapshot();
      });

      it('should update angular.json', async () => {
        const content = appTree.readContent('/angular.json');
        expect(content).toMatchSnapshot();
      });
    });

    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync(
            'ng-add',
            { ...defaultFeatureOptions, lazy: false },
            appTree
          )
          .toPromise();
      });

      it('should import appropriate modules', async () => {
        const orderApprovalModule = appTree.readContent(
          orderApprovalFeatureModulePath
        );
        expect(orderApprovalModule).toContain(
          `import { OrderApprovalRootModule } from "@spartacus/organization/order-approval/root";`
        );
        expect(orderApprovalModule).toContain(
          `import { OrderApprovalModule } from "@spartacus/organization/order-approval";`
        );
      });

      it('should not contain lazy loading syntax', async () => {
        const orderApprovalModule = appTree.readContent(
          orderApprovalFeatureModulePath
        );
        expect(orderApprovalModule).not.toContain(
          `import('@spartacus/organization/order-approval').then(`
        );
      });
    });

    describe('lazy loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', defaultFeatureOptions, appTree)
          .toPromise();
      });

      it('should import OrderApprovalRootModule and contain the lazy loading syntax', async () => {
        const orderApprovalModule = appTree.readContent(
          orderApprovalFeatureModulePath
        );
        expect(orderApprovalModule).toContain(
          `import { OrderApprovalRootModule, ORGANIZATION_ORDER_APPROVAL_FEATURE } from "@spartacus/organization/order-approval/root";`
        );
        expect(orderApprovalModule).toContain(
          `import('@spartacus/organization/order-approval').then(`
        );
      });

      it('should not contain the OrderApprovalModule import', () => {
        const orderApprovalModule = appTree.readContent(
          orderApprovalFeatureModulePath
        );
        expect(orderApprovalModule).not.toContain(
          `import { OrderApprovalModule } from "@spartacus/organization/order-approval";`
        );
      });
    });

    describe('i18n', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', defaultFeatureOptions, appTree)
          .toPromise();
      });

      it('should import the i18n resource and chunk from assets', async () => {
        const orderApprovalModule = appTree.readContent(
          orderApprovalFeatureModulePath
        );
        expect(orderApprovalModule).toContain(
          `import { orderApprovalTranslationChunksConfig, orderApprovalTranslations } from "@spartacus/organization/order-approval/assets";`
        );
      });
      it('should provideConfig', async () => {
        const orderApprovalModule = appTree.readContent(
          orderApprovalFeatureModulePath
        );
        expect(orderApprovalModule).toContain(
          `resources: orderApprovalTranslations,`
        );
        expect(orderApprovalModule).toContain(
          `chunks: orderApprovalTranslationChunksConfig,`
        );
      });
    });
  });

  describe('b2b features', () => {
    beforeEach(async () => {
      appTree = await schematicRunner
        .runSchematicAsync('ng-add', defaultFeatureOptions, appTree)
        .toPromise();
    });

    it('configuration should be added', () => {
      const configurationModule = appTree.readContent(
        `src/app/spartacus/${SPARTACUS_CONFIGURATION_MODULE}.module.ts`
      );
      expect(configurationModule).toMatchSnapshot();
    });

    it('should update package.json', () => {
      const packageJson = appTree.readContent(`package.json`);
      expect(packageJson).toContain(SPARTACUS_SETUP);
    });
  });
});
