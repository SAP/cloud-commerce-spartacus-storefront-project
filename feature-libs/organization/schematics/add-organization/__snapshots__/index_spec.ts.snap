// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus Organization schematics: ng-add Administration feature styling should create a proper scss file 1`] = `"@import \\"@spartacus/organization\\";"`;

exports[`Spartacus Organization schematics: ng-add Administration feature styling should update angular.json 1`] = `
"{
  \\"$schema\\": \\"./node_modules/@angular/cli/lib/config/schema.json\\",
  \\"version\\": 1,
  \\"newProjectRoot\\": \\"\\",
  \\"projects\\": {
    \\"schematics-test\\": {
      \\"projectType\\": \\"application\\",
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      },
      \\"root\\": \\"\\",
      \\"sourceRoot\\": \\"src\\",
      \\"prefix\\": \\"app\\",
      \\"architect\\": {
        \\"build\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:browser\\",
          \\"options\\": {
            \\"outputPath\\": \\"dist/schematics-test\\",
            \\"index\\": \\"src/index.html\\",
            \\"main\\": \\"src/main.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.app.json\\",
            \\"aot\\": true,
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\"
            ],
            \\"styles\\": [
              \\"src/styles.scss\\",
              \\"src/styles/spartacus/organization.scss\\"
            ],
            \\"scripts\\": []
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"fileReplacements\\": [
                {
                  \\"replace\\": \\"src/environments/environment.ts\\",
                  \\"with\\": \\"src/environments/environment.prod.ts\\"
                }
              ],
              \\"optimization\\": true,
              \\"outputHashing\\": \\"all\\",
              \\"sourceMap\\": false,
              \\"extractCss\\": true,
              \\"namedChunks\\": false,
              \\"extractLicenses\\": true,
              \\"vendorChunk\\": false,
              \\"buildOptimizer\\": true,
              \\"budgets\\": [
                {
                  \\"type\\": \\"initial\\",
                  \\"maximumWarning\\": \\"2mb\\",
                  \\"maximumError\\": \\"5mb\\"
                },
                {
                  \\"type\\": \\"anyComponentStyle\\",
                  \\"maximumWarning\\": \\"6kb\\",
                  \\"maximumError\\": \\"10kb\\"
                }
              ]
            }
          }
        },
        \\"serve\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:dev-server\\",
          \\"options\\": {
            \\"browserTarget\\": \\"schematics-test:build\\"
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"browserTarget\\": \\"schematics-test:build:production\\"
            }
          }
        },
        \\"extract-i18n\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:extract-i18n\\",
          \\"options\\": {
            \\"browserTarget\\": \\"schematics-test:build\\"
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:karma\\",
          \\"options\\": {
            \\"main\\": \\"src/test.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.spec.json\\",
            \\"karmaConfig\\": \\"karma.conf.js\\",
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\"
            ],
            \\"styles\\": [
              \\"src/styles.scss\\",
              \\"src/styles/spartacus/organization.scss\\"
            ],
            \\"scripts\\": []
          }
        },
        \\"lint\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:tslint\\",
          \\"options\\": {
            \\"tsConfig\\": [
              \\"tsconfig.app.json\\",
              \\"tsconfig.spec.json\\",
              \\"e2e/tsconfig.json\\"
            ],
            \\"exclude\\": [
              \\"**/node_modules/**\\"
            ]
          }
        },
        \\"e2e\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:protractor\\",
          \\"options\\": {
            \\"protractorConfig\\": \\"e2e/protractor.conf.js\\",
            \\"devServerTarget\\": \\"schematics-test:serve\\"
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"devServerTarget\\": \\"schematics-test:serve:production\\"
            }
          }
        }
      }
    }
  },
  \\"defaultProject\\": \\"schematics-test\\"
}"
`;

exports[`Spartacus Organization schematics: ng-add Order approval feature styling should create a proper scss file 1`] = `"@import \\"@spartacus/organization\\";"`;

exports[`Spartacus Organization schematics: ng-add Order approval feature styling should update angular.json 1`] = `
"{
  \\"$schema\\": \\"./node_modules/@angular/cli/lib/config/schema.json\\",
  \\"version\\": 1,
  \\"newProjectRoot\\": \\"\\",
  \\"projects\\": {
    \\"schematics-test\\": {
      \\"projectType\\": \\"application\\",
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      },
      \\"root\\": \\"\\",
      \\"sourceRoot\\": \\"src\\",
      \\"prefix\\": \\"app\\",
      \\"architect\\": {
        \\"build\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:browser\\",
          \\"options\\": {
            \\"outputPath\\": \\"dist/schematics-test\\",
            \\"index\\": \\"src/index.html\\",
            \\"main\\": \\"src/main.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.app.json\\",
            \\"aot\\": true,
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\"
            ],
            \\"styles\\": [
              \\"src/styles.scss\\",
              \\"src/styles/spartacus/organization.scss\\"
            ],
            \\"scripts\\": []
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"fileReplacements\\": [
                {
                  \\"replace\\": \\"src/environments/environment.ts\\",
                  \\"with\\": \\"src/environments/environment.prod.ts\\"
                }
              ],
              \\"optimization\\": true,
              \\"outputHashing\\": \\"all\\",
              \\"sourceMap\\": false,
              \\"extractCss\\": true,
              \\"namedChunks\\": false,
              \\"extractLicenses\\": true,
              \\"vendorChunk\\": false,
              \\"buildOptimizer\\": true,
              \\"budgets\\": [
                {
                  \\"type\\": \\"initial\\",
                  \\"maximumWarning\\": \\"2mb\\",
                  \\"maximumError\\": \\"5mb\\"
                },
                {
                  \\"type\\": \\"anyComponentStyle\\",
                  \\"maximumWarning\\": \\"6kb\\",
                  \\"maximumError\\": \\"10kb\\"
                }
              ]
            }
          }
        },
        \\"serve\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:dev-server\\",
          \\"options\\": {
            \\"browserTarget\\": \\"schematics-test:build\\"
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"browserTarget\\": \\"schematics-test:build:production\\"
            }
          }
        },
        \\"extract-i18n\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:extract-i18n\\",
          \\"options\\": {
            \\"browserTarget\\": \\"schematics-test:build\\"
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:karma\\",
          \\"options\\": {
            \\"main\\": \\"src/test.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.spec.json\\",
            \\"karmaConfig\\": \\"karma.conf.js\\",
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\"
            ],
            \\"styles\\": [
              \\"src/styles.scss\\",
              \\"src/styles/spartacus/organization.scss\\"
            ],
            \\"scripts\\": []
          }
        },
        \\"lint\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:tslint\\",
          \\"options\\": {
            \\"tsConfig\\": [
              \\"tsconfig.app.json\\",
              \\"tsconfig.spec.json\\",
              \\"e2e/tsconfig.json\\"
            ],
            \\"exclude\\": [
              \\"**/node_modules/**\\"
            ]
          }
        },
        \\"e2e\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:protractor\\",
          \\"options\\": {
            \\"protractorConfig\\": \\"e2e/protractor.conf.js\\",
            \\"devServerTarget\\": \\"schematics-test:serve\\"
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"devServerTarget\\": \\"schematics-test:serve:production\\"
            }
          }
        }
      }
    }
  },
  \\"defaultProject\\": \\"schematics-test\\"
}"
`;

exports[`Spartacus Organization schematics: ng-add b2b features configuration should be added 1`] = `
"import { NgModule } from '@angular/core';
import { translationChunksConfig, translations } from \\"@spartacus/assets\\";
import { FeaturesConfig, I18nConfig, OccConfig, provideConfig, SiteContextConfig } from \\"@spartacus/core\\";
import { defaultB2bCheckoutConfig, defaultB2bOccConfig } from \\"@spartacus/setup\\";
import { defaultCmsContentProviders, layoutConfig, mediaConfig } from \\"@spartacus/storefront\\";

@NgModule({
  declarations: [],
  imports: [
  ],
  providers: [provideConfig(layoutConfig), provideConfig(mediaConfig), ...defaultCmsContentProviders, provideConfig(<OccConfig>{
    backend: {
      occ: {
        baseUrl: 'https://localhost:9002',
      }
    },
  }), provideConfig(<SiteContextConfig>{
    context: {
      currency: ['USD'],
      language: ['en'],
    },
  }), provideConfig(<I18nConfig>{
    i18n: {
      resources: translations,
      chunks: translationChunksConfig,
      fallbackLang: 'en'
    },
  }), provideConfig(<FeaturesConfig>{
    features: {
      level: '3.2'
    }
  }), provideConfig(defaultB2bOccConfig), provideConfig(defaultB2bCheckoutConfig)]
})
export class SpartacusConfigurationModule { }
"
`;
